scene/main/node.cpp

```diff
@@ -2203,49 +2136,40 @@ void Node::_duplicate_signals(const Node *p_original, Node *p_copy) const {
 	}
 }
 
-Node *Node::duplicate_and_reown(const Map<Node *, Node *> &p_reown_map) const {
-
-	ERR_FAIL_COND_V(get_filename() != "", NULL);
-
-	Node *node = NULL;
-
-	Object *obj = ClassDB::instance(get_class());
-	ERR_EXPLAIN("Node: Could not duplicate: " + String(get_class()));
-	ERR_FAIL_COND_V(!obj, NULL);
-	node = Object::cast_to<Node>(obj);
-	if (!node)
-		memdelete(obj);
-	ERR_FAIL_COND_V(!node, NULL);
-
-	node->set_name(get_name());
-
-	List<PropertyInfo> plist;
+void Node::_duplicate_owners(Node *p_copy, const Map<Node *, Node *> &p_reown_map) const {
+	Node *owner = get_owner();
 
-	get_property_list(&plist);
+	if (p_reown_map.has(owner))
+		owner = p_reown_map[owner];
 
-	for (List<PropertyInfo>::Element *E = plist.front(); E; E = E->next()) {
+	if (owner) {
+		NodePath p = get_path_to(owner);
+		if (owner != this) {
+			Node *new_owner = p_copy->get_node(p);
+			if (new_owner) {
+				p_copy->set_owner(new_owner);
+			}
+		}
+	}
 
-		if (!(E->get().usage & PROPERTY_USAGE_STORAGE))
-			continue;
-		String name = E->get().name;
-		node->set(name, get(name));
+	for (int i = 0; i < get_child_count(); i++) {
+		get_child(i)->_duplicate_owners(p_copy->get_child(i), p_reown_map);
 	}
+}
 
-	List<GroupInfo> groups;
-	get_groups(&groups);
+Node *Node::duplicate_and_reown(const Map<Node *, Node *> &p_reown_map) const {
 
-	for (List<GroupInfo>::Element *E = groups.front(); E; E = E->next())
-		node->add_to_group(E->get().name, E->get().persistent);
+	ERR_FAIL_COND_V(get_filename() != "", NULL);
 
-	for (int i = 0; i < get_child_count(); i++) {
+	Node *node = _duplicate(DUPLICATE_SIGNALS | DUPLICATE_GROUPS | DUPLICATE_SCRIPTS);
 
-		get_child(i)->_duplicate_and_reown(node, p_reown_map);
-	}
+	_duplicate_owners(node, p_reown_map);
```

Comment:
Does it work well to do this as a two-step operation, first duplicate all, then reown all? Asking because `_duplicate_and_reown` used to have this, so it seems pretty strict on the call order and ownership:
```
	if (get_owner() != get_parent()->get_owner())
		return;
```